### PR TITLE
Add ADR proposal for error responses

### PR DESCRIPTION
**Pull request by malte-laukoetter**
RISDEV-4013

### CODE REVIEW

File: doc/adr/0013-error-responses.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+[RFC-9457 (Problem Details for HTTP APIs)](https://www.rfc-editor.org/rfc/rfc9457) addresses how such information can be
+sent by an API.
+
+## Decision
+
+We follow [RFC-9457 (Problem Details for HTTP APIs)](https://www.rfc-editor.org/rfc/rfc9457) for our error responses.
+All error responses are JSON objects.
+
+The `type` field always starts with `/errors/`. Therefore, all errors are relative to our application.
```
Comment (hannes-mk) [inline doc/adr/0013-error-responses.md]: maybe "relative URLs"

File: doc/adr/0013-error-responses.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+For errors related to a specific norm the `instance` field is filled with the eli of the norm starting with `/eli/`.
+
+Both `title` and `details` are written in english.
+
```
Comment (hannes-mk) [inline doc/adr/0013-error-responses.md]: Maybe add what is required and what is optional.

File: doc/adr/0013-error-responses.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+### Multiple error messages in one response
+
+For errors that can have multiple messages (e.g. after a validation) we mostly follow the example from Section 3 of
+RFC-9457 for `HTTP/1.1 422 Unprocessable Content`.
+
+The multiple errors MUST be of the same general type. Otherwise, only the most severe error is reported. E.g. it is
+possible to send multiple `norm-not-valid` errors but not a `norm-not-valid` error at the same time as
+a `norm-not-found` error.
+
```
Comment (hannes-mk) [inline doc/adr/0013-error-responses.md]: so, in this case, only `norm-not-found` is returned

File: doc/adr/0013-error-responses.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+#### LegalDocML.de
+
+- `eid`: an eId (LegalDocML.de) referencing the part of the norm responsible for the error (preferred)
+- `guid`: a GUID (LegalDocML.de) referencing the part of the norm responsible for the error (optionally or as
+  alternative to the eid)
+
+### Displaying messages for the user
+
+To create messages to display to the user the `type` must be used in combination with the extension members. The `title`
```
Comment (hannes-mk) [inline doc/adr/0013-error-responses.md]: "extension members": maybe "other fields"

File: doc/adr/0013-error-responses.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+RFC-9457 for `HTTP/1.1 422 Unprocessable Content`.
+
+The multiple errors MUST be of the same general type. Otherwise, only the most severe error is reported. E.g. it is
+possible to send multiple `norm-not-valid` errors but not a `norm-not-valid` error at the same time as
+a `norm-not-found` error.
+
+The response includes an additional `errors` array. The elements of the `errors` array can have the same fields as the
+global object. If no `type` is provided it is assumed that it is the same as the parent type.
+
+For both JSON and LegalDocML.de some additional fields are defined to reference parts of a JSON-object or LDML.de
```
Comment (VictorDelCampo) [inline doc/adr/0013-error-responses.md]: I wasn't sure what you meant here with "JSON" but now I understand you mean the request payload. All these fields "pointer", "eid" and "guid" are extension members we already have in mind that we will be using.

I would maybe rephrase that part and maybe say something like possible/probable extension members, and then you name them and explain for what would they be used (specially JSON being the request payload :) )

File: doc/adr/0013-error-responses.md (hunk 4 above)
Comment (VictorDelCampo) [inline doc/adr/0013-error-responses.md]: I guess the frontend would then use the `type` which will probably contain parameters matching the extension members to build the message with those right?

File: doc/adr/0013-error-responses.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+Example error response for validation:
+
+```
+{
+ "type": "/errors/norm-not-valid",
+ "title": "Norm validation failed.",
+ "instance": "/eli/bund/bgbl-1/2017/s419/2017-03-15/1/deu/regelungstext-1"
+ "errors": [{
+    "type": "/errors/norm-not-valid/quoted-structure/target-and-up-to-are-not-siblings
```
Comment (VictorDelCampo) [inline doc/adr/0013-error-responses.md]: I guess you left out the `title` in the sub-error because you were just doing an example? This brings me back to the point from @hannes-mk saying to specify which properties would be required (like `type`) and which can be optional (`title`, `details`, `instance` and extension members)?

**Review by VictorDelCampo — COMMENTED** [review]
Nice work Malte. I like this concept so much! It is going some steps further than what was proposed in the caselaw team. I just have some minor comments you could use to extend the ADR if you feel like to

File: doc/adr/0013-error-responses.md (hunk 4 above)
Comment (malte-laukoetter) [reply doc/adr/0013-error-responses.md]: yes exactly thats the idea

File: doc/adr/0013-error-responses.md (hunk 5 above)
Comment (malte-laukoetter) [reply doc/adr/0013-error-responses.md]: i moved this to its own section, so i hope it is clearer now?

File: doc/adr/0013-error-responses.md (hunk 6 above)
Comment (malte-laukoetter) [reply doc/adr/0013-error-responses.md]: I adjusted the examples to better align with the text and added some sentences about which fields are required and recommended :)

**Comment by VictorDelCampo** [comment]
@malte-laukoetter looks great! Just adapt the date and can be merged :)
